### PR TITLE
ENYO-6334: Fix marquee text alignment on restart

### PR DESCRIPTION
**Pull request by ryanjduffy**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Uses IntersectionObserver on the spacer to calculate the subpixel offset missed by `scrollWidth` and adjust the width by that amount.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

**Comment by webOS101** [comment]
I'm seeing an issue when using `forceDirection="ltr"` when using RTL content.  The same problem does not occur on `develop` (though, it's not quite right there, either).

**Comment by Sangwook-Lee** [comment]
LGTM

File: packages/ui/Marquee/MarqueeBase.js [hunk 1]
```diff
@@ -218,12 +231,11 @@ const MarqueeBase = kind({
 					style={clientStyle}
 					onTransitionEnd={onMarqueeComplete}
 				>
-					{children}
+					<span>{children}</span>
```
[the remark this answers is not in the document]
Comment (ryanjduffy) [reply packages/ui/Marquee/MarqueeBase.js]: I tried that but noticed more animation latency when that was included.

**Comment by ryanjduffy** [comment]
Note that we've found issues with this approach in other contexts (e.g. in Button) so we can't proceed with it as is. I've started another attempt in https://github.com/enactjs/enact/tree/feature/ENYO-6334-ryan and a third locally that isn't shareable yet.

File: packages/ui/Marquee/MarqueeBase.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xbinary) [reply packages/ui/Marquee/MarqueeBase.js]: The reason I mentioned about `span` is that I have been reported with the following issue.

an App uses the following code in real case.
`<Heading showLine />` // no children
Space was created by span tag that did not exist previously.

So if there is no children, how about removing span tag?

[the remark this answers is not in the document]
Comment (ryanjduffy) [reply packages/ui/Marquee/MarqueeBase.js]: That's fair. If we take this approach or some variation, I'll include support for that scenario

**Pull request closed without merging**